FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -366,8 +366,14 @@ private async Task RenewJobLockLeaseAsync(CancellationToken cancellationToken =
             }
             catch
             {
-                _logger.LogError("Failed to renew job lease.");
-                throw;
+                if (_renewLockTimer.Enabled)
+                {
+                    _logger.LogError("Failed to renew job lease.");
+                    throw;
+                }
+
+                _logger.LogInformation("Refresh timer already stopped.");
+                return;
```

Comment:
how the caller handle this case that renew locker fails while no exception is threw?

Reply:
This is some corner cases when the timer stops but the renew request was not cancelled